public/app/features/datasources/components/EditDataSource.tsx

```diff
@@ -149,6 +151,23 @@ export function EditDataSourceView({
     );
   }
 
+  // Used by extensions that can only modify certain parts of the `jsonData`
+  const onUpdateJsonData = (jsonData: Record<string, unknown>) => {
+    const protectedFields = ['authType', 'defaultRegion', 'profile', 'manageAlerts', 'alertmanagerUid'];
+    onOptionsChange({ ...dataSource, jsonData: { ...dataSource.jsonData, ...omit(jsonData, protectedFields) } });
+  };
+
+  const { extensions } = getPluginElementExtensions({
+    extensionPointId: PluginExtensionPoints.DataSourceConfig,
+    context: { jsonData: dataSource.jsonData, onUpdateJsonData, pluginId: dataSource.type },
+  });
+
+  // HEADS UP!
+  // The "myorg-basic-app" is just for testing purposes and should be removed before merging this PR, or before this PR becomes ready-for-review.
+  const allowedExtensions = extensions.filter(({ pluginId }) =>
+    ['grafana-pdc-app', 'myorg-basic-app'].includes(pluginId)
```

Comment:
Any chance to add "grafana-auth-app"? This is the second use case when we would like to give users a way to generate tokens when editing data sources...

Reply:
Sure 👍